src/main/groovy/oe/espresso/latte/SettingsSet.groovy

```diff
@@ -66,4 +66,19 @@ class SettingsSet<E> implements Set<E> {
     Object[] toArray() {
         this.defaults.toArray() + this.settings.toArray()
     }
+
+    /**
+        iterates over all the default settings,
+        and the current settings
+    */
+    Set<E> each(Closure closure) {
+        def all = []
+        all.addAll defaults
+        all.addAll settings
+        all.unique(false).each(closure);
```

Comment:
what does `unique(false)` mean? shouldn't the `all` set have unique values only?

Reply:
i changed this to not use unique.  problem is toArray() isn't using it, so dups won't get filtered out and i prefer to keep behavior consistent between each and toarray.

please review my changes.

to answer your question unique(false) prevents mutation of the underlying set.  unique(true) in this case would eliminate dups from "unique" rather than returning a new set

Reply:
what would test against "default" set to?  the constructors don't allow having "no defaults".

Reply:
you can't do "new SettingsSet()". You always have to do new SettingsSet([] as Set)

Reply:
we can't get duplicates.  when you use ([] as Set) groovy uses a SET which enforces uniqueness.  the concatenation in toArray() could in theory violate this but the "contains()" function should be preventing this from happening.